### PR TITLE
Add support for listing and altering consumer offsets to the AdminClient

### PR DESCRIPTION
**Pull request by regispl**
As mentioned in Discord, I was playing a bit with the new functionality in the Kafka Admin Client added in 2.5 (listing and altering consumer offsets) and implemented the support for them to zio-kafka for fun.

One note: I defined enums using `sealed abstract class` - this is my personal preference (if I'm not using enumeratum) and I saw it's used in the code, but I also saw `sealed trait` being used elsewhere, so please let me know if you prefer me to change it.

### CODE REVIEW

**Comment by svroonland** [comment]
Hi @regispl, is there anything we can do to help to complete this PR?

**Comment by regispl** [comment]
@svroonland Take my kids away for a day 😉 But seriously, I had some issues with the type of the ZIO Environment in the test that I tried to add, that I couldn't understand at that time. I didn't have time to read up on the topic and fix it. I hope to have some time to revisit it soon. If I can't I may just commit the incomplete test and ask for help.

**Comment by svroonland** [comment]
Feel free to push what you have and I'll see what I can do to help!

**Comment by regispl** [comment]
It took a bit of wrestling and getting up to speed with ZStream, ZLayer and few other parts of the ecosystem, but I've finally found some time over the weekend and managed to add tests (I admit, they're not testing a multitude of scenarios - they're quite simplistic). 

This PR should now be complete and ready for review.

File: src/main/scala/zio/kafka/admin/AdminClient.scala [hunk 1]
```diff
@@ -193,6 +242,80 @@ object AdminClient {
...
+    }
+
+    final case object ReadCommitted extends IsolationLevel {
+      override def asJava = JIsolationLevel.READ_COMMITTED
+    }
+  }
+
+  case class ListOffsetsOptions(
+    isolationLevel: IsolationLevel = IsolationLevel.ReadUncommitted,
+    timeoutMs: Option[Int]
```
Comment (svroonland) [inline src/main/scala/zio/kafka/admin/AdminClient.scala:286]: If we're having a scala wrapper type anyway, we should use the more scala-idiomatic `scala.duration.Duration` type here and below.

**Review by svroonland — COMMENTED** [review]
Nice work, thank for this! A few nitpicks and then this should be ready to merge.

File: src/main/scala/zio/kafka/admin/AdminClient.scala [hunk 2]
```diff
@@ -131,6 +144,43 @@ case class AdminClient(private val adminClient: JAdminClient) {
...
+    }
+  }.map(_.asScala.toMap.bimap(TopicPartition(_), ListOffsetsResultInfo(_)))
+
+  /**
+   * Alter offsets for the specified partitions and consumer group.
+   */
+  def alterConsumerGroupOffsets(
+    groupId: String,
+    offsets: Map[TopicPartition, OffsetAndMetadata],
+    alterConsumerGroupOffsetsOptions: Option[AlterConsumerGroupOffsetsOptions] = None
```
[the remark this answers is not in the document]
Comment (regispl) [reply src/main/scala/zio/kafka/admin/AdminClient.scala:171]: Naming is important and APIs can be tricky to evolve once defined, so it's definitely a valid comment and I wouldn't call it nitpicking 👍 

On the other hand, however, please note that all other functions in this class use the same naming pattern for options (`describeConfigsOptions`, `createPartitionsOptions`, `deleteTopicsOptions` and 2 or 3 others) which seem to map 1:1 to Java client names, so there's the aspect of the naming consistency involved here as well, that perhaps should be considered as well. 

Let me know if you still want me to update it or should we keep it as is - I'm happy with both 👍

File: src/main/scala/zio/kafka/admin/AdminClient.scala [hunk 3]
```diff
@@ -131,6 +144,43 @@ case class AdminClient(private val adminClient: JAdminClient) {
       )
     }
   }
+
+  /**
+   * List offset for the specified partitions.
+   */
+  def listOffsets(
+    topicPartitionOffsets: Map[TopicPartition, OffsetSpec],
+    listOffsetOptions: Option[ListOffsetsOptions] = None
```
[the remark this answers is not in the document]
Comment (regispl) [reply src/main/scala/zio/kafka/admin/AdminClient.scala:153]: Same as https://github.com/zio/zio-kafka/pull/201#discussion_r503072976

File: src/main/scala/zio/kafka/admin/AdminClient.scala (hunk 1 above)
Comment (regispl) [reply src/main/scala/zio/kafka/admin/AdminClient.scala:286]: Good spot, will fix, thanks 👍

Comment (regispl) [reply src/main/scala/zio/kafka/admin/AdminClient.scala:286]: Actually, I just noticed that there's also `zio.duration` - should I perhaps use this one?

File: src/main/scala/zio/kafka/admin/AdminClient.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (svroonland) [reply src/main/scala/zio/kafka/admin/AdminClient.scala:171]: Makes sense, let's keep it like this for now. I'll add an issue to rename the options parameters in a separate PR.

File: src/main/scala/zio/kafka/admin/AdminClient.scala (hunk 1 above)
Comment (svroonland) [reply src/main/scala/zio/kafka/admin/AdminClient.scala:286]: Yes, I think that's better. `zio.duration.Duration` is a type alias for `java.time.Duration`, but there's some extension methods and conversions.

**Review by svroonland — APPROVED** [review]
Great!

**Pull request merged**